test/passphrase/validation.js

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright © 2017 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import {
+	countCapitalCharacters,
+	countPassphraseWhitespaces,
+	countPassphraseWords,
+	validatePassphrase,
+} from '../../src/passphrase/validation';
+
+describe('passphrase validation', () => {
+	describe('countPassphraseWhitespaces', () => {
```

Comment:
Do we really need to export and test these simple helper functions separately to the main validation function? It seemed like they were extracted for readability rather than because they abstract away some serious logic.

Reply:
I thought better have more than less tests regarding this logic. Since it can be quite sensitive.

Reply:
As per discussion, different clients may have different requirements for passphrase validation and these functions are exported for their convenience (and hence require unit tests).